components/scream/scripts/gen_boiler.py

```diff
@@ -0,0 +1,1878 @@
+        lambda *x           : "The cxx kokkos function declaration(<name>)"
+    )),
+
+    ("cxx_incl_impl", (
+        lambda phys, sub, gb: "{}_functions.hpp".format(phys),
+        lambda phys, sub, gb: expect_exists(phys, sub, gb, "cxx_incl_impl"),
+        lambda phys, sub, gb: re.compile(r"^\s*#\s*endif\s+//\s*KOKKOS_ENABLE_CUDA"), # insert at end of impl includes, reqs special comment
+        lambda phys, sub, gb: re.compile(r'^\s*#\s*include\s+"{}"'.format(get_piece_data(phys, sub, "cxx_func_impl", FILEPATH, gb))),
+        lambda phys, sub, gb: re.compile(r".*"),
+        lambda *x           : "The include of *impl.hpp file at bottom of main hpp"
+    )),
+
+    ("cxx_func_impl", (
+        lambda phys, sub, gb: "{}_{}_impl.hpp".format(phys, sub),
+        lambda phys, sub, gb: create_template(phys, sub, gb, "cxx_func_impl"),
+        lambda phys, sub, gb: get_namespace_close_regex(phys),   # insert at end of namespace
+        lambda phys, sub, gb: get_cxx_function_begin_regex(sub, template="Functions<S,D>"), # cxx begin
+        lambda phys, sub, gb: get_cxx_close_block_regex(at_line_start=True), # terminating }
+        lambda *x           : "The cxx kokkos function stub implementation(<name>)"
+    )),
+
+    ("cxx_bfb_unit_decl", (
+        lambda phys, sub, gb: "tests/{}_unit_tests_common.hpp".format(phys),
+        lambda phys, sub, gb: expect_exists(phys, sub, gb, "cxx_bfb_unit_decl"),
+        lambda phys, sub, gb: get_cxx_close_block_regex(semicolon=True), # insert at end of test struct
+        lambda phys, sub, gb: get_cxx_struct_begin_regex(get_data_test_struct_name(sub)), # struct decl
+        lambda phys, sub, gb: re.compile(r".*;\s*$"), # end of struct decl
+        lambda *x           : "The cxx unit test struct declaration"
+    )),
+
+    ("cxx_bfb_unit_impl", (
+        lambda phys, sub, gb: "tests/{}_{}_tests.cpp".format(phys, sub),
+        lambda phys, sub, gb: create_template(phys, sub, gb, "cxx_bfb_unit_impl"),
+        lambda phys, sub, gb: get_cxx_close_block_regex(semicolon=True, at_line_start=True), # insert of end of struct
+        lambda phys, sub, gb: get_cxx_function_begin_regex("run_bfb", static=True),  # run_bfb
+        lambda phys, sub, gb: get_cxx_close_block_regex(comment="run_bfb"), # } // run_bfb # reqs special comment
+        lambda *x           : "The cxx bfb unit test implementation"
+    )),
+
+    ("cxx_eti", (
+        lambda phys, sub, gb: "{}_{}.cpp".format(phys, sub),
+        lambda phys, sub, gb: create_template(phys, sub, gb, "cxx_eti"),
+        lambda phys, sub, gb: re.compile(".*"), # insert at top of file
+        lambda phys, sub, gb: re.compile(".*"), # start at top of file
+        lambda phys, sub, gb: get_namespace_close_regex("scream"), #end of file
+        lambda *x           : "The cxx explicit template instatiation file"
+    )),
+
+    ("cmake_impl_eti", (
+        lambda phys, sub, gb: "CMakeLists.txt",
```

Comment:
Reading the code I was having a hard time telling what sub, and gb were. After some time I realized gb=gen-boiler, and sub=soubroutine. Right?

Could we use 'func' instead of 'sub', or spell it out completely (subroutine)? The word 'sub' is quite vague until you link it to subroutine, and code maintainers (other than Jim) might have a hard time diving into the script.

Reply:
That's right. I'm conflicted about using longer names since I like that all these lambda are one line and long names would either make these lines really long or force me to introduce newlines.

Reply:
This is not a big deal, just trying to make the life of the maintainer who comes after you a bit easier.

What aboud `func`? I know functions and subroutines are different things in f90, but `sub` was not very clear at first. Or maybe `subrtn`? Anything that makes it unequivocal what we're taking about, I guess.

About long names: these names are only to be used inside this script, so it only affects the maintainer(s) of this tool.

Anyhow, as I said, not a big deal.